extensions/internal-dictionary/src/index.tsx

```diff
@@ -0,0 +1,307 @@
+      if (!result.success) {
+        const { error } = result;
+        showToast({
+          title: "Schema Validation Failed",
+          message:
+            "Errors: " +
+            error.issues.map((issue) => issue.path[issue.path.length - 1] + ": " + issue.message).join(", "),
+          style: Toast.Style.Failure,
+        });
+        return;
+      }
+
+      const { data: parsedItems } = result;
+      parsedItems!.map((item) => LocalStorage.setItem(item.id, JSON.stringify(item)));
+
+      LocalStorage.allItems().then((storageEntries) => {
+        const parsedEntries: DictionaryEntry[] = Object.entries(storageEntries).map(([_, e]) => JSON.parse(e));
+        setSearchQuery("");
+        setEntries(parsedEntries);
+        showToast({
+          title: "Import Successful",
+          message: "Imported " + parsedItems.length + " item" + (parsedItems.length > 1 ? "s" : ""),
+          style: Toast.Style.Success,
+        });
+      });
+    });
+  };
+
+  const bulkActions = (
+    <ActionPanel.Section title="Clipboard Actions">
+      <Action
+        title="Export All"
+        shortcut={{ modifiers: ["cmd", "shift"], key: "y" }}
+        icon={Icon.Clipboard}
+        onAction={() => {
+          Clipboard.copy(JSON.stringify(entries, undefined, 2));
+          showToast({ title: "Copied All Entries" });
+        }}
+      />
+      <Action
+        title="Import From"
+        icon={Icon.ArrowDownCircle}
+        shortcut={{ modifiers: ["cmd", "shift"], key: "p" }}
+        onAction={() => tryImportFromClipboard()}
+      />
+    </ActionPanel.Section>
+  );
+
+  return (
+    <List
+      searchBarPlaceholder="Searching entries"
+      onSearchTextChange={(txt) => setSearchQuery(txt)}
+      actions={
+        <ActionPanel title="Actions">
+          <CreateDictionaryEntryAction onCreate={handleCreate} />
+          {bulkActions}
+        </ActionPanel>
+      }
+    >
+      <List.Section title="Results" subtitle={displayEntries?.length + ""}>
+        {(displayEntries || []).map((searchResult) => {
+          return (
+            <SearchListItem
+              key={searchResult.id}
+              searchResult={searchResult!}
+              handleCreate={handleCreate}
+              handleEdit={handleEdit}
+              handleDelete={handleDelete}
+              bulkActions={bulkActions}
+            />
+          );
+        })}
+      </List.Section>
+    </List>
+  );
+}
+
+function SearchListItem({
+  searchResult,
+  handleCreate,
+  handleEdit,
+  handleDelete,
+  bulkActions,
+}: {
+  searchResult: DictionaryEntry;
+  handleCreate: (entry: DictionaryEntry) => void;
+  handleEdit: (entry: DictionaryEntry) => void;
+  handleDelete: (entry: DictionaryEntry) => void;
+  bulkActions: ReactNode;
+}) {
+  return (
+    <List.Item
+      title={searchResult.title}
+      subtitle={searchResult.description}
+      accessories={searchResult.score ? [{ text: (1 - searchResult.score).toFixed(2) }] : []}
```

Comment:
Does this make sense to show to the user? At least we could add a tooltip explaining what it is 🙂